Lib/typing.py

```diff
@@ -2020,14 +2024,15 @@ def _proto_hook(cls, other):
                 break
 
             # ...or in annotations, if it is a sub-protocol.
-            if (
-                issubclass(other, Generic)
-                and getattr(other, "_is_protocol", False)
-                and attr in _lazy_annotationlib.get_annotations(
-                    base, format=_lazy_annotationlib.Format.FORWARDREF
-                )
-            ):
-                break
+            if issubclass(other, Generic) and getattr(other, "_is_protocol", False):
+                try:
+                    annos = base.__annotations__
+                except Exception:
+                    annos = _lazy_annotationlib.get_annotations(
+                        base, format=_lazy_annotationlib.Format.FORWARDREF
+                    )
+                if attr in annos:
+                    break
```

Comment:
I don't mind this change _too_ much, but is it _necessary_ to keep the `annotationlib` import lazy? I'm not sure that `_proto_hook` is executed when `typing` is imported?

The previous code is a bit simpler. Though I suppose the new code might be a bit faster? Not sure...

Reply:
It's not necessary, but it seemed better to defer the cost of importing annotationlib from another case where you don't need it.

Though this only matters if you do `issubclass()` between two protocols which has got to be rather rare.

Reply:
I see, fair enough. I think a comment wouldn't go amiss to say why we're going for the slightly more complex code here! (Not to speed up typing.py import time, but because the import itself could be expensive, and most of the time it's unnecessary)